test/concurrent/executor/test_cached_thread_pool.rb

```diff
@@ -0,0 +1,14 @@
+$LOAD_PATH.unshift File.expand_path("../../../lib", __dir__)
```

Comment:
Unfortunately, testing the functionality is only possible with minitest.
You can see the comment surround [disable_at_exit_handlers!](https://ruby-concurrency.github.io/concurrent-ruby/1.1.4/Concurrent.html#disable_at_exit_handlers!-class_method) to read why it can't be tested in RSpec.

For now I've included a single raw minitest file -- I'm open to suggestion as to how best to incorporate it into the Rakefile according to this build process.

Reply:
It would seem better to spin off a separate process and ensure that it terminates within some timeout. The test is fine but I think it should run in its own subprocess.

Reply:
There were tests that essentially did that in RSpec -- the failure however was that Minitest was messing with the AtExit handlers.
Given that it's a very popular test framework; I thought it would be reasonable to incorporate it sparingly for such cases. I'm not too strongly attached one way or another.

Reply:
Could we instead start a simple process that messes with the `AtExit` handlers? That might be easier to understand and we wouldn't introduce another development dependency. WDYT?

Reply:
A sub-process could be written which calls `disable_at_exit_handlers!`
Personal opinion on the matter though is that a test that verifies that it works with minitest is pretty useful given the  popularity of the gem.

This PR has seem some churn -- I'd love to get this fix in and follow-up with ammendments/improvements to the testing facility also ?

That would also play well with the idea I've been chatting about on _gitter_ channel about even potentially removing **AtExit** completely. (It's uneeded now if all the threads are daemon)